inference/core/workflows/core_steps/transformations/detection_filter.py

```diff
@@ -173,49 +175,80 @@ def get_manifest(cls) -> Type[WorkflowBlockManifest]:
 
     async def run_locally(
         self,
-        predictions: List[List[dict]],
+        predictions: Union[List[Dict[str, Any]], List[sv.Detections]],
         filter_definition: Union[
             DetectionFilterDefinition, CompoundDetectionFilterDefinition
         ],
         image_metadata: List[dict],
         prediction_type: List[str],
-    ) -> Union[List[Dict[str, Any]], Tuple[List[Dict[str, Any]], FlowControl]]:
+    ) -> Union[
+        Union[List[sv.Detections], List[Dict[str, Any]]],
+        Tuple[Union[List[sv.Detections], List[Dict[str, Any]]], FlowControl],
+    ]:
         filter_callable = build_filter_callable(definition=filter_definition)
-        result_predictions, result_parent_id = [], []
+        result_predictions, result_parent_ids = [], []
         for detections in predictions:
-            filtered_prediction = [
-                deepcopy(detection)
-                for detection in detections
-                if filter_callable(detection)
-            ]
-            result_predictions.append(filtered_prediction)
-            result_parent_id.append(
-                [prediction[PARENT_ID_KEY] for prediction in filtered_prediction]
-            )
+            if isinstance(detections, list):
```

Comment:
[question] in which context this is still list?

Reply:
I thought about supporting predictions produced by `inference` too but upon reflection I remove this.